recipes/base64/all/conanfile.py

```diff
@@ -0,0 +1,95 @@
+import os
+from conans import ConanFile, AutoToolsBuildEnvironment, CMake, tools
+
+
+class CppcodecConan(ConanFile):
+    name = "base64"
+    license = "BSD-2-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/aklomp/base64"
+    description = "Fast Base64 stream encoder/decoder"
+    topics = ("base64", "codec")
+    exports_sources = ["CMakeLists.txt", "cmake/*", "test/*"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "fPIC": True,
+    }
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+        del self.settings.compiler.cppstd
+        del self.settings.compiler.libcxx
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+        if self.settings.compiler == "Visual Studio":
+            os.rename("CMakeLists.txt", os.path.join(self._source_subfolder, "CMakeLists.txt"))
+            os.rename("cmake", os.path.join(self._source_subfolder, "cmake"))
+            os.rename(os.path.join("test", "CMakeLists.txt"), os.path.join(self._source_subfolder, "test", "CMakeLists.txt"))
+        else:
+            s = "lib/libbase64.o: $(OBJS)"
+            tools.replace_in_file(os.path.join(self._source_subfolder, "Makefile"), s, """
+all: lib/libbase64.a
+lib/libbase64.a: lib/libbase64.o
+	$(AR) rc $@ $<
+""" + s)
```

Comment:
My main remark is to add the cmake build script in one patch (like https://patch-diff.githubusercontent.com/raw/aklomp/base64/pull/7.patch)
The `Makefile` change can also be moved to a patch and applied unconditionally (even when buliding using Make, because it is not used).